### PR TITLE
Added handling testing option for get-block-hash.

### PR DESCRIPTION
**Pull request by orizi**
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5368)

### CODE REVIEW

**Review by ArniStarkware — CHANGES_REQUESTED** [review]
> Reviewed 2 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5368)* status: 2 of 4 files reviewed, 3 unresolved discussions (waiting on @ArielElp, @gilbens-starkware, and @orizi)

___
*[`corelib/src/starknet/testing.cairo` line 77 at r1](https://reviewable.io/reviews/starkware-libs/cairo/5368#-NuwdaiO-2JC-csR6zCe:-NuwdaiO-2JC-csR6zCf:b-3d06m2) ([raw file](https://github.com/starkware-libs/cairo/blob/1ed04996968197ec9a6af0800fbfba77bb07986c/corelib/src/starknet/testing.cairo#L77)):*
> ```Cairo
> pub fn set_block_hash(block_id: u64, value: felt252) {
>     cheatcode::<'set_block_hash'>(array![block_id.into(), value].span());
> }
> ```

Note `block id` and `block number` are different.
Not that I understand what `block id` means in the context of this repo.

Block id - is the address of the block saved to airospike.
Block number - the sequential number of the block in the chain.

Maybe the names were mixed somewhere in the process?



_Suggestion:_
```Cairo
/// Set the hash for a block.
/// Unset blocks values call would fail.
pub fn set_block_hash(block_number: u64, value: felt252) {
    cheatcode::<'set_block_hash'>(array![block_number.into(), value].span());
}
```
___
*[`crates/cairo-lang-runner/src/casm_run/mod.rs` line 143 at r1](https://reviewable.io/reviews/starkware-libs/cairo/5368#-NuwcK4k1q9oxsFGBH-n:-NuwcK4k1q9oxsFGBH-o:b851c8m) ([raw file](https://github.com/starkware-libs/cairo/blob/7e6367d082ae88a22996b9a976569e1bc3e7f1ce/crates/cairo-lang-runner/src/casm_run/mod.rs#L143)):*
> ```XML
>     /// The simulated execution info.
>     exec_info: ExecutionInfo,
>     next_id: Felt252,
> ```

This value is unrelated to `get_block_hash`, is it?



_Code quote:_
```XML
    next_id: Felt252,
```
___
*[`crates/cairo-lang-runner/src/casm_run/mod.rs` line 144 at r1](https://reviewable.io/reviews/starkware-libs/cairo/5368#-Nuwc8A78brrt27uDeKJ:-Nuwc8A78brrt27uDeKK:b-o42bb8) ([raw file](https://github.com/starkware-libs/cairo/blob/1ed04996968197ec9a6af0800fbfba77bb07986c/crates/cairo-lang-runner/src/casm_run/mod.rs#L144)):*
> ```XML
>     exec_info: ExecutionInfo,
>     /// The test provided block hashes.
>     block_hash: HashMap<u64, Felt252>,
> ```

Is this runner used for something other than tests? The docstring indicates it is not.



_Suggestion:_
```XML
    /// A mock history, mapping block number to the class hash.
    block_hash: HashMap<u64, Felt252>,
```

**Comment by ArniStarkware** [comment]
*[`corelib/src/starknet/testing.cairo` line 77 at r1](https://reviewable.io/reviews/starkware-libs/cairo/5368#-NuwdaiO-2JC-csR6zCe:-NuwfV9vCEE1cM5vmSK3:b6xkqa1) ([raw file](https://github.com/starkware-libs/cairo/blob/1ed04996968197ec9a6af0800fbfba77bb07986c/corelib/src/starknet/testing.cairo#L77)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

Note `block id` and `block number` are different.
Not that I understand what `block id` means in the context of this repo.

Block id - is the address of the block saved to airospike.
Block number - the sequential number of the block in the chain.

Maybe the names were mixed somewhere in the process?
</blockquote></details>

I see they were only mixed here.

TL;DR: rename `block_id` -> `block_number` PLZ.

**Review by orizi — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5368)* status: 2 of 4 files reviewed, 2 unresolved discussions (waiting on @ArielElp, @ArniStarkware, and @gilbens-starkware)

___
*[`corelib/src/starknet/testing.cairo` line 77 at r1](https://reviewable.io/reviews/starkware-libs/cairo/5368#-NuwdaiO-2JC-csR6zCe:-NuwgURe6OIhczopoU30:b-896fix) ([raw file](https://github.com/starkware-libs/cairo/blob/1ed04996968197ec9a6af0800fbfba77bb07986c/corelib/src/starknet/testing.cairo#L77)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

I see they were only mixed here.

TL;DR: rename `block_id` -> `block_number` PLZ.
</blockquote></details>

Done.
___
*[`crates/cairo-lang-runner/src/casm_run/mod.rs` line 143 at r1](https://reviewable.io/reviews/starkware-libs/cairo/5368#-NuwcK4k1q9oxsFGBH-n:-NuwgV-JB0grJx4EGsZD:bb1h2ld) ([raw file](https://github.com/starkware-libs/cairo/blob/7e6367d082ae88a22996b9a976569e1bc3e7f1ce/crates/cairo-lang-runner/src/casm_run/mod.rs#L143)):*
<details><summary><i>Previously, ArniStarkware (Arnon Hod) wrote…</i></summary><blockquote>

This value is unrelated to `get_block_hash`, is it?
</blockquote></details>

removing in pre-PR.

**Review by ArniStarkware — COMMENTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5368)* status: 2 of 4 files reviewed, all discussions resolved (waiting on @ArielElp and @gilbens-starkware)

**Review by gilbens-starkware — APPROVED** [review]
:lgtm:

> Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5368#-:-Nux5e0W8v7JohNjvZEQ:bnfp4nl)* status: all files reviewed (commit messages unreviewed), all discussions resolved (waiting on @ArielElp)

**Review by gilbens-starkware — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5368)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ArielElp)

**Review by ArniStarkware — APPROVED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5368)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ArielElp)
